src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -312,6 +312,47 @@ impl ScalarFunc {
         self.children[i as usize].eval_string(ctx, row)
     }
 
+    #[inline]
+    pub fn field_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        // Pattern matching made explicit here.
+        // As per the MySQL doc, if the first argument is NULL, this function always returns 0.
+        let s = match self.children[0].eval_string(ctx, row) {
```

Comment:
I think we should `eval_int` here?
BTW, we can do it like `match self.children[0].eval_int(ctx, row)?`.

Reply:
Nope. All the arguments of the `FIELD()` function are strings as per [MySQL doc](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_field).

Reply:
It seems that I misunderstood the meaning. I'll fix that.